src/index.ts

```diff
@@ -1,130 +0,0 @@
-import {configLogs, error, getErrorMessage, info} from "./utils/logger";
-import {Config, loadConfig} from "./utils/configFileReader";
-import {PullRequestInfo} from "./pullRequestInfo";
-import {PipelineRun} from "./utils/piplineRun";
-import {Deployment, KubernetesAPI} from "./utils/kubernetes";
-
-const args = process.argv.slice(2);
-const configFileIndex = args.findIndex(arg => arg === '-p');
-let config: Config | undefined;
-
-const pjson = require('../package.json');
-const version = pjson.version;
-
-if (configFileIndex !== -1 && configFileIndex + 1 < args.length) {
-	const configFilePath = args[configFileIndex + 1];
-	info(`Using config file path: ${configFilePath}`, '');
-
-	config = loadConfig(configFilePath);
-	if (!config) {
-		throw new Error("Error loading config file")
-	}
-	configLogs(config);
-} else {
-	throw new Error("Config file not specified")
-}
-
-info(`wp-continuous-integration started with version ${version}`, {});
-
-async function scheduleActivePRsToDeployments() {
-	const pullRequests = await PullRequestInfo.getAvailablePRsSortedByDate(config!);
-	const expiredPullRequests = await PullRequestInfo.getExpiredPRs(config!);
-	const deployments = await KubernetesAPI.getDeploymentsSortedByLastDeployDesc(config!.NAMESPACE);
-	await Promise.all(deployments.map(dep => scheduleToDeployment(config!.NAMESPACE, dep, pullRequests, [], deployments, true)));
-	await Promise.all(deployments.map(dep => scheduleToDeployment(config!.NAMESPACE, dep, pullRequests, expiredPullRequests, deployments, false)));
-	for (const pr of pullRequests) {
-		await pr.createComment('🍍 ' + pr.skipped())
-	}
-}
-
-async function scheduleToDeployment(
```

Comment:
The part that picks the `Deployment` needs to be synchronous so as to avoid [TOCTTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) bugs (such as two different calls deciding to use the same `Deployment`). As currently written, this may or may not be the case; it's hard to tell. It shouldn't. According to previous comment, please refactor to eliminate the concern of picking the `Deployment` out of this function (as well as the arguments needed to reach that decision; having more than 3–4 function parameters is a code smell.)